intro.js

```diff
@@ -1079,8 +1099,8 @@
       tooltipLayer.appendChild(progressLayer);
 
       //add helper layer number
-      if (this._options.showStepNumbers == true) {
-        var helperNumberLayer = document.createElement('span');
+      var helperNumberLayer = document.createElement('span');
+      if (this._options.showStepNumbers === true) {
```

Comment:
Why is it necessary?

Reply:
I can't remember this one in context, but I believe it was something like this:

```javascript
if (1 === 1) {
  var a = 3;
}
// a is defined within an if-block
return a;
```

While it technically "works", it's bad style and makes it more ambiguous; generally you want to define all your variables at the top of the block-scope that you'll be using them in; at the top of functions, at the top of if statements, at the top of forloops, and don't use them outside of those block scopes.  I think it's good practice for readability if nothing else.